### PR TITLE
Add CIS 1.8

### PR DESCRIPTION
**Pull request by orouz**
### Summary of your changes

Adds CIS GCP 1.8

| failed (same user both admin and user role) |  passed (user doesn't have both admin and user role)  |
|-----------|---|
| Screenshot 2023-07-31 at 15 10 53  | Screenshot 2023-07-31 at 15 11 07|

 

### Related Issues

- https://github.com/elastic/security-team/issues/6179

### Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?
- [x] Generate rule metadata using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rules-metadata)
- [x] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

> **Note** Make sure to bump cloudbeat [`version/policy.go`](https://github.com/elastic/cloudbeat/blob/main/version/policy.go#L20) after merging this PR and drafting a new release.

### CODE REVIEW

File: bundle/compliance/cis_gcp/rules/cis_1_8/rule.rego [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+members_with_both_roles := {m |
+	role_a_binding = data_adapter.iam_policy.bindings[_]
+	role_b_binding = data_adapter.iam_policy.bindings[_]
+	role_a_binding.role == "roles/iam.serviceAccountAdmin"
+	role_b_binding.role == "roles/iam.serviceAccountUser"
+	m = role_a_binding.members[_]
+	m = role_b_binding.members[_]
+}
+
+default is_valid = false
```
Comment (uri-weisman) [inline bundle/compliance/cis_gcp/rules/cis_1_8/rule.rego:16]: move to top

**Review by oren-zohar — CHANGES_REQUESTED** [review]

**Comment by orouz** [comment]
`Test Policies / Update rules status (pull_request)` was reverted an is not on `main`. no idea why github is running it

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: bundle/compliance/cis_gcp/rules/cis_1_8/rule.rego [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import data.compliance.lib.common
+import data.compliance.policy.gcp.data_adapter
+import future.keywords.if
+
+default is_valid = false
+
+is_valid if {
+	count(members_with_both_roles) == 0
+}
```
[the remark this answers is not in the document]
Comment (orouz) [reply bundle/compliance/cis_gcp/rules/cis_1_8/rule.rego]: removed

File: bundle/compliance/cis_gcp/rules/cis_1_8/rule.rego [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+finding = result if {
+	data_adapter.is_iam_service_account
+	data_adapter.has_policy
+
+	result := common.generate_result_without_expected(
+		common.calculate_result(is_valid),
+		data_adapter.iam_policy,
+	)
+}
```
[the remark this answers is not in the document]
Comment (orouz) [reply bundle/compliance/cis_gcp/rules/cis_1_8/rule.rego]: removed `is_valid` in favor of `no_admin_with_user_role` which is more descriptive to the actual rule. also moved out of global rule scope

File: bundle/compliance/cis_gcp/rules/cis_1_8/rule.rego [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+members_with_both_roles := {m |
+	role_a_binding = data_adapter.iam_policy.bindings[_]
+	role_b_binding = data_adapter.iam_policy.bindings[_]
+	role_a_binding.role == "roles/iam.serviceAccountAdmin"
+	role_b_binding.role == "roles/iam.serviceAccountUser"
+	m = role_a_binding.members[_]
+	m = role_b_binding.members[_]
+}
```
[the remark this answers is not in the document]
Comment (orouz) [reply bundle/compliance/cis_gcp/rules/cis_1_8/rule.rego]: renamed bindings to `admin` and `user` and re-arranged to make it less puzzling

File: bundle/compliance/cis_gcp/rules/cis_1_8/rule.rego [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+finding = result if {
+	data_adapter.is_cloud_resource_manager_project
+	data_adapter.has_policy
+
+	is_valid := count(members_with_both_roles) == 0
```
[the remark this answers is not in the document]
Comment (orouz) [reply bundle/compliance/cis_gcp/rules/cis_1_8/rule.rego]: good point! renamed to `no_admin_with_user_role`

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**